### PR TITLE
ch4: default MPIR_CVAR_CH4_ROOTS_ONLY_PMI on

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description
Using ROOTS_ONLY_PMI significantly improves the large-scale init time. It is about time to make it default.

[skip warnings]

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [ ] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by hzhou** [comment]
test:mpich/ch4/all

**Comment by hzhou** [comment]
test:mpich/warnings

**Comment by hzhou** [comment]
test:mpich/ch4/all

**Comment by hzhou** [comment]
test:mpich/ch4/most

**Comment by hzhou** [comment]
test:mpich/ch4/xpmem

**Comment by hzhou** [comment]
test:mpich/ch4/ofi

File: src/mpid/ch4/netmod/ofi/init_addrxchg.c [hunk 1]
```diff
@@ -155,6 +166,14 @@ int MPIDI_OFI_addr_exchange_root_ctx(void)
         MPIDU_bc_table_destroy();
     }
 
+    /* check */
+    if (MPIDI_OFI_ENABLE_AV_TABLE) {
+        for (int r = 0; r < size; r++) {
+            MPIDI_OFI_addr_t *av ATTRIBUTE((unused)) = &MPIDI_OFI_AV(&MPIDIU_get_av(0, r));
+            MPIR_Assert(av->dest[0][0] == get_root_av_table_index(r));
```
[the remark this answers is not in the document]
Comment (hzhou) [reply src/mpid/ch4/netmod/ofi/init_addrxchg.c:173]: The `av` is the entry associated with the rank `r`.

[the remark this answers is not in the document]
Comment (hzhou) [reply src/mpid/ch4/netmod/ofi/init_addrxchg.c:173]: The reason we do this checking is explained in the comments at the front of the file. We compressed the dimensions of the av table based on the assumption that all the remote av are inserted exactly with the same order on each local context (vci, nic), thus they should have exactly the same fi_addr. The check confirms this assumption is still correct.

[the remark this answers is not in the document]
Comment (raffenet) [reply src/mpid/ch4/netmod/ofi/init_addrxchg.c:173]: OK. `av->dest[0][0]` has type `fi_addr_t` if I'm not mistaken? And we are comparing that with an integer rank?

[the remark this answers is not in the document]
Comment (hzhou) [reply src/mpid/ch4/netmod/ofi/init_addrxchg.c:173]: Yes, it is an integer under `MPIDI_OFI_ENABLE_AV_TABLE`

**Review by raffenet — APPROVED** [review]

**Comment by hzhou** [comment]
Thanks for the review @raffenet . The tests are not clean, but all of them are testing congestion issue that are not related to this PR. I'll merge now. If there are recurring testing issues, we'll address them afterward.

**Pull request merged**